cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -376,7 +377,8 @@ public void apply(MetadataDataset input) throws Exception {
   public MetadataSearchResponse search(String namespaceId, String searchQuery,
                                        Set<EntityTypeSimpleName> types,
                                        SortInfo sortInfo, int offset, int limit,
-                                       int numCursors, String cursor, boolean showHidden) throws BadRequestException {
+                                       int numCursors, String cursor, boolean showHidden,
+                                       Set<EntityScope> entityScope) throws BadRequestException {
```

Comment:
I'm not suggesting you change this, but wondering why you have this as `Set<EntityScope>`, instead of `EnumSet<EntityScope>`